CS410Assignment2/src/sensor/TemperatureSensorAdapter.java

```diff
@@ -0,0 +1,52 @@
+package sensor;
+
+import java.awt.Color;
+
+public class TemperatureSensorAdapter implements SensorAdapter{
+	
+	private TemperatureSensor adaptee;
+	
+	public TemperatureSensorAdapter(TemperatureSensor t) {
+		this.adaptee = t;
+	}
+	
+	//TemperatureSensor t = new TemperatureSensor();
+
+	@Override
+	public String getReport() {
+		return adaptee.getTempReport();
+	}
+
+	@Override
+	public Double getValue() {
+		return adaptee.senseTemperature();
+	}
+
+	@Override
+	public String getType() {
+		return adaptee.getSensorType();
+	}
+	
+	@Override
+	public Color getColor() {
```

Comment:
The issue I encountered where the value and color wouldn't line up you solved in a different way, however the end result is the same. Each sensors get value->sensetemp/readpressure/readradiation method is only called once, and the color is determined from that single value.